static-instrumenter/agent-instrumenter/build.gradle.kts

```diff
@@ -1,12 +1,138 @@
+  }
+
+  val isolateJavaagentLibs by registering(Copy::class) {
+    dependsOn(relocateJavaagentLibs)
+    isolateClasses(relocateJavaagentLibs.get().outputs.files)
+    into("$buildDir/isolated/javaagentLibs")
+  }
+
+  shadowJar {
+    configurations = listOf(javaagent, bootstrapLibs)
+
+    dependsOn(isolateJavaagentLibs)
+    from(isolateJavaagentLibs.get().outputs)
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+
+    manifest {
+      attributes.put("Main-Class", "io.opentelemetry.contrib.staticinstrumenter.agent.main.Main")
+      attributes.put("Agent-Class", "io.opentelemetry.contrib.staticinstrumenter.agent.OpenTelemetryStaticAgent")
+      attributes.put("Premain-Class", "io.opentelemetry.contrib.staticinstrumenter.agent.OpenTelemetryStaticAgent")
+      attributes.put("Can-Redefine-Classes", "true")
+      attributes.put("Can-Retransform-Classes", "true")
+      attributes.put("Implementation-Vendor", "OpenTelemetry")
+      attributes.put("Implementation-Version", "demo-${project.version}")
+    }
+  }
+
+  val mainShadowJar by registering(Jar::class) {
+    archiveFileName.set("opentelemetry-static-agent.jar")
+    from(zipTree(shadowJar.get().archiveFile))
+
+    manifest {
+      attributes(shadowJar.get().manifest.attributes)
+    }
+  }
+
+  // TODO: the resulting jar contains both inst and unpacked inst,
+  //  but we need only the unpacked inst here
+  val createNoInstAgent by registering(ShadowJar::class) {
+
+    configurations = listOf(javaagent)
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+
+    inputs.files.forEach {
+      if (it.name.endsWith(".jar")) {
+        from(zipTree(it)) {
+          include("inst/")
+          eachFile {
+            // drop inst/
+            relativePath = RelativePath(true, *relativePath.segments.drop(1).toTypedArray())
+            rename("(.*)\\.classdata\$", "\$1.class")
+          }
+        }
+      }
+    }
+    archiveFileName.set("opentelemetry-no-inst-agent.jar")
+  }
+
+  assemble {
+    dependsOn(shadowJar, mainShadowJar, createNoInstAgent)
+  }
+
+  val copyJarsToTestRes by registering(Copy::class) {
+    dependsOn(assemble)
+    from("$buildDir/libs/opentelemetry-no-inst-agent.jar")
+    from("$buildDir/libs/opentelemetry-static-agent.jar")
+    destinationDir = File("src/integrationTest/resources")
+  }
+
+  val integrationTest by existing {
+    dependsOn(copyJarsToTestRes)
+  }
+}
+
+fun CopySpec.isolateClasses(jars: Iterable<File>) {
+  jars.forEach {
+    from(zipTree(it)) {
+      into("inst")
+      rename("^(.*)\\.class\$", "\$1.classdata")
+      // Rename LICENSE file since it clashes with license dir on non-case sensitive FSs (i.e. Mac)
+      rename("""^LICENSE$""", "LICENSE.renamed")
+      exclude("META-INF/INDEX.LIST")
+      exclude("META-INF/*.DSA")
+      exclude("META-INF/*.SF")
+    }
+  }
+}
+
+tasks.withType<ShadowJar>().configureEach {
+  relocate("io.opentelemetry.instrumentation", "io.opentelemetry.javaagent.shaded.instrumentation")
 }
```

Comment:
It's because `io.opentelemetry.instrumentation.api.config.Config` is used in `agent-extension`, eg [here](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/319/files#diff-ca9deb6e10bef4c5710b60496a0519716cf8b45356a7b8ff84aeab89ce05e1c7R23-R24)

Reply:
Can you add a comment explaining why we're relocating? (i.e. because of instrumentation-api)